### PR TITLE
refactor(tsc)：TSC access adjustment

### PR DESCRIPTION
**Pull request by wzh425**
# Description

refactor：TSC access adjustment

### CODE REVIEW

File: src/Infrastructure/Masa.Mc.Infrastructure.Tsc/ObservabilityExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Mc.Infrastructure.Tsc;
+
+public static class ObservabilityExtensions
```
Comment (doddgu) [inline src/Infrastructure/Masa.Mc.Infrastructure.Tsc/ObservabilityExtensions.cs:6]: I think this extension need add to TSC SDK.

File: src/Services/Masa.Mc.Service/Domain/MessageRecords/Aggregates/MessageRecord.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) MASA Stack All rights reserved.
 // Licensed under the Apache License. See LICENSE.txt in the project root for license information.
 
+using Masa.BuildingBlocks.Data;
```
Comment (doddgu) [inline src/Services/Masa.Mc.Service/Domain/MessageRecords/Aggregates/MessageRecord.cs:4]: Move it to _Imports.

File: src/Services/Masa.Mc.Service/appsettings.json [hunk 3]
```diff
@@ -34,9 +34,10 @@
...
   },
-  "OTLP": {
-    "Endpoint": "http://10.10.90.114:32147",
+  "MasaTscOption": {
     "ServiceName": "masa-mc-service",
+    "ServiceNameSpace": "Development",
     "ServiceVersion": "1.0.0"
-  }
+  },
+  "OtlpUrl": "http://otel.lonsid.cn:32247"
```
Comment (doddgu) [inline src/Services/Masa.Mc.Service/appsettings.json:42]: Is this internal domain?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Services/Masa.Mc.Service/appsettings.json (hunk 3 above)
Comment (wzh425) [reply src/Services/Masa.Mc.Service/appsettings.json]: Yes, other environment configuration files have been configured. This file does not need to be removed now

File: src/Infrastructure/Masa.Mc.Infrastructure.Tsc/ObservabilityExtensions.cs (hunk 1 above)
Comment (wzh425) [reply src/Infrastructure/Masa.Mc.Infrastructure.Tsc/ObservabilityExtensions.cs:6]: I think so, too

File: src/Services/Masa.Mc.Service/Domain/MessageRecords/EventHandler/UpdateMessageRecordUserEventHandler.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+        if (messageRecord == null)
+        {
+            _logger.LogInformation($"UpdateMessageRecordUserEventHandler:messageRecords is null");
+            return;
+        }
+
+        var user = await GetMessageRecordsUser(messageRecord);
+
+        if (user == null)
```
Comment (doddgu) [inline src/Services/Masa.Mc.Service/Domain/MessageRecords/EventHandler/UpdateMessageRecordUserEventHandler.cs:34]: user ??= await CreateExternalUserAsync(messageRecord);

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Services/Masa.Mc.Service/Domain/MessageRecords/EventHandler/UpdateMessageRecordUserEventHandler.cs (hunk 4 above)
Comment (wzh425) [reply src/Services/Masa.Mc.Service/Domain/MessageRecords/EventHandler/UpdateMessageRecordUserEventHandler.cs]: When the sent user is not in auth, call the create external user interface to create the user

Comment (wzh425) [reply src/Services/Masa.Mc.Service/Domain/MessageRecords/EventHandler/UpdateMessageRecordUserEventHandler.cs]: 1

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**